kitsune/sumo/static/js/wiki_search.js

```diff
@@ -0,0 +1,78 @@
+(function($, _) {
+  var searchTimeout;
+  var locale = $('html').attr('lang');
+
+  var $searchField = $('#search-related');
+  var $relatedDocsList = $('#related-docs-list');
+  var $resultsList;
+
+  // To search for only wiki articles we pass w=1
+  var search = new k.Search('/' + locale + '/search', {w: 1, format: 'json'});
+
+  function createResultsList() {
+    $resultsList = $('<div />').addClass('input-dropdown');
+    $searchField.after($resultsList);
+    $resultsList.show();
+
+    $resultsList.on('click', '[data-pk]', function() {
+      var $this = $(this);
+
+      $relatedDocsList.children('.empty-message').remove();
+
+      if (!$relatedDocsList.children('[data-pk=' + $this.data('pk') + ']').length) {
+        var context = {
+          name: 'related_documents',
+          doc: {
+            id: $this.data('pk'),
+            title: $this.text()
+          }
+        };
+
+        $relatedDocsList.append(_.render('wiki-related-doc.html', context));
+      }
+    });
+  }
+
+  function showResults(data) {
+    if (!$resultsList) {
+      createResultsList();
+    }
+    $resultsList.html(_.render('wiki-search-results.html', data));
+  }
+
+  function handleSearch() {
+    var $this = $(this);
+    if ($this.val().length === 0) {
+      if (searchTimeout) {
+        window.clearTimeout(searchTimeout);
+      }
+      $resultsList.html('');
```

Comment:
Sometimes I get `$resultsList is undefined` from this line. I think mainly when the window closes right after I make a search.